typescript/infra/scripts/agents/update-agent-config.ts

```diff
@@ -235,7 +235,7 @@ export async function writeAgentConfig(
   const filepath = getAgentConfigJsonPath(envNameToAgentEnv[environment]);
   console.log(`Writing config to ${filepath}`);
   if (fs.existsSync(filepath)) {
-    const currentAgentConfig: AgentConfig = readJSONAtPath(filepath);
+    const currentAgentConfig: AgentConfig = readJson<AgentConfig>(filepath);
```

Comment:
When we use it like this, there's a false impression that the file contents is validate. Should we add validation? Or remove that sort of typing from the function to underline the fact that the type-cast is just that?